presto-spi/src/main/java/com/facebook/presto/spi/type/NamedTypeSignature.java

```diff
@@ -25,20 +25,32 @@
 public class NamedTypeSignature
 {
     private final Optional<String> name;
+    private final boolean delimited;
     private final TypeSignature typeSignature;
 
     @JsonCreator
     public NamedTypeSignature(
             @JsonProperty("name") Optional<String> name,
+            @JsonProperty("delimited") boolean delimited,
             @JsonProperty("typeSignature") TypeSignature typeSignature)
     {
         this.name = requireNonNull(name, "name is null");
+        if (!name.isPresent() && delimited) {
+            throw new IllegalArgumentException("delimited but name is absent");
```

Comment:
This sanity check here indicates a deficiency in the API. I believe it makes more sense to have a `Optional<FieldName>` field in the class instead of an `Optional<String>` and a `boolean`. The `FieldName` class will have two fields: `name` and `delimited`.